src/archival/utils.rs

```diff
@@ -59,6 +74,27 @@ pub async fn is_row_exists(pool: &PgPool, row_id: i32) -> bool {
     }
 }
 
+pub async fn make_archival_network_request(
+    url: &str,
+    endpoint_url: &str,
+) -> Result<ArchivalResponse, ArchivalError> {
+    let client = Arc::clone(&REQWEST_CLIENT);
+    let response = client
+        .post(endpoint_url)
+        .body(format!("url={}", url))
+        .send()
+        .await?;
+    let response_text = response.text().await?;
+    // Success response, contains job_id
+    if let Ok(res) = serde_json::from_str::<ArchivalResponse>(&response_text) {
+        return Ok(res);
+    }
+    // HTML response, case when IA can not archive URL, and is under maintenance
+    Ok(ArchivalResponse::Html(ArchivalHtmlResponse {
+        html: response_text,
+    }))
```

Comment:
I think it would be better to condense the `Err` and `Html` error directly into `ArchivalError` here.

Reply:
Done ✅ , Check: https://github.com/yellowHatpro/mb-exurl-ia-service/commit/6d7ce4b88d9321775ebee303541ad11896032502